examples/applications/plot_time_series_lagged_features.py

```diff
@@ -0,0 +1,433 @@
+# because we expect that the number of rentals cannot be 100% accurately
+# predicted from the features. It can be influenced by other variables
+# not properly captured by the existing lagged features. For instance
+# whether or not it will rain in the next hour cannot be fully be anticipated
+# from the past hours bike rental data. This is what we call aleatoric
+# uncertainty.
+#
+# Quantile regression makes it possible to give a finer description of that
+# distribution without making strong assumptions on its shape.
+#
+# The conditional 5th percentile (a.k.a. 0.05-quantile) can be estimated with:
+gbrt_percentile_05 = HistGradientBoostingRegressor(loss="quantile", quantile=0.05)
+evaluate(gbrt_percentile_05, X, y, cv=ts_cv)
+
+# %%
+# The conditional median (0.50-quantile) can be estimated with:
+gbrt_median = HistGradientBoostingRegressor(loss="quantile", quantile=0.5)
+evaluate(gbrt_median, X, y, cv=ts_cv)
+
+# %%
+# And finally the 0.95 quantile:
+gbrt_percentile_95 = HistGradientBoostingRegressor(loss="quantile", quantile=0.95)
+evaluate(gbrt_percentile_95, X, y, cv=ts_cv)
+
+# %%
+# A Qualitative Look at the Predictions
+# -------------------------------------
+# We can now visualize the performance of the model with regards
+# to the 5th percentile, median and the 95th percentile:
+all_splits = list(ts_cv.split(X, y))
+train_idx, test_idx = all_splits[0]
+
+X_train, X_test = X.iloc[train_idx], X.iloc[test_idx]
+y_train, y_test = y.iloc[train_idx], y.iloc[test_idx]
+
+max_iter = 50
+gbrt_mean_poisson = HistGradientBoostingRegressor(loss="poisson", max_iter=max_iter)
+gbrt_mean_poisson.fit(X_train, y_train)
+mean_predictions = gbrt_mean_poisson.predict(X_test)
+
+gbrt_median = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.5, max_iter=max_iter
+)
+gbrt_median.fit(X_train, y_train)
+median_predictions = gbrt_median.predict(X_test)
+
+gbrt_percentile_5 = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.05, max_iter=max_iter
+)
+gbrt_percentile_5.fit(X_train, y_train)
+percentile_5_predictions = gbrt_percentile_5.predict(X_test)
+
+gbrt_percentile_95 = HistGradientBoostingRegressor(
+    loss="quantile", quantile=0.95, max_iter=max_iter
+)
+gbrt_percentile_95.fit(X_train, y_train)
+percentile_95_predictions = gbrt_percentile_95.predict(X_test)
+
+last_hours = slice(-96, None)
+fig, ax = plt.subplots(figsize=(15, 7))
+plt.title("Predictions by regression models")
+ax.plot(
+    y_test.values[last_hours],
+    "x-",
+    alpha=0.2,
+    label="Actual demand",
+    color="black",
+)
+ax.plot(
+    median_predictions[last_hours],
+    "^-",
+    label="GBRT median",
+)
+ax.plot(
+    mean_predictions[last_hours],
+    "x-",
+    label="GBRT mean (Poisson)",
+)
+ax.fill_between(
+    np.arange(96),
+    percentile_5_predictions[last_hours],
+    percentile_95_predictions[last_hours],
+    alpha=0.3,
+    label="GBRT 90% interval",
+)
+_ = ax.legend()
+

```

Comment:
After each plot we need a paragraph to explain how to analyze what is displayed. Here it's interesting to notice that the blue area between the 5% and 95% percentile estimators has a width that varies with the time of the day:

- at night, the blue band is much narrower: the pair of models is quite certain that there will be a small number of bike rentals. And furthermore those seem correct in the sense that the actual demand stays in that blue band;
- during the day, the blue band is much wider: the uncertainty grows, probably because of the variability of the weather that can have a very large impact, especially on a week-end day;
- we can also see that during week-days, the commute pattern is still visible in the 5% and 95% estimations;
- finally, it is expected that 10% of the time, the actual demand does not lie between the 5% and 95% percentile estimates. On this test span, it the actual demand seems to be higher, especially during the rush ours. It might reveal that our 95% percentile estimator underestimates the demand peaks. This could be be quantitatively confirmed by computing empirical coverage numbers as done in this section of another example that we should link from this time-series forecasting example:

https://scikit-learn.org/stable/auto_examples/ensemble/plot_gradient_boosting_quantile.html#calibration-of-the-confidence-interval